### PR TITLE
Restructures code base for public/private includes

### PR DESCRIPTION
**Pull request by jlaura**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -30,21 +38,17 @@ set_target_properties(sensorutils PROPERTIES
...
 
 target_include_directories(sensorutils
                            PUBLIC
-                           ${SENSORUTILS_INCLUDE_DIRS}
+                           ${SENSORUTILS_PUBLIC_INCLUDE_DIRS}
+                           PRIVATE
+                           ${SENSORUTILS_PRIVATE_INCLUDE_DIRS}
                            ${ARMADILLO_INCLUDE_DIRS}
-)
+                           ${CSM_INCLUDE_DIR})
```
Comment (jlaura) [inline CMakeLists.txt]: This is the major change. We now have public and private headers.

File: api/SensorModel.h [hunk 2]
```diff
@@ -17,7 +17,7 @@ class SensorModel {
 *
 */
 public:
-
+  ~SensorModel() { };
```
Comment (jlaura) [inline api/SensorModel.h]: Good form to include or okay to omit?

File: include/SensorMath.h [hunk 3]
```diff
@@ -1,22 +1,14 @@
...
-using namespace arma;
 
 namespace sensormath {
 
-  vec cartesianToVec(CartesianPoint point);
-  vec imageToVec(ImagePoint point);
-
-  CartesianVector vecToCartesian(vec vec); 
-  ImagePoint vecToImage(vec vec);
-
```
Comment (jlaura) [inline include/SensorMath.h]: I did the namespace thing here. This has implications for the tests. I comment on those below.

File: src/sensormath/SensorMath.cpp [hunk 4]
```diff
@@ -3,11 +3,12 @@
...
 #include <cmath>
 
+#include <armadillo>
+
 using namespace std;
 using namespace arma;
 
-namespace sensormath {
-
+namespace {
```
Comment (jlaura) [inline src/sensormath/SensorMath.cpp]: Here is that anonymous namespace.

File: tests/CMakeLists.txt [hunk 5]
```diff
@@ -11,8 +11,13 @@ add_executable(runSensorUtilsTests TestMain.cpp
...
 
-target_link_libraries(runSensorUtilsTests PUBLIC sensorutils ${GMOCK_LIBRARY} pthread)
-
+target_link_libraries(runSensorUtilsTests PUBLIC 
+                      sensorutils 
+                      ${GMOCK_LIBRARY} 
+                      pthread)
+include_directories(${SENSORUTILS_PRIVATE_INCLUDE_DIRS}
+                    ${SENSORUTILS_PUBLIC_INCLUDE_DIRS}
+                    ${ARMADILLO_INCLUDE_DIR})
```
Comment (jlaura) [inline tests/CMakeLists.txt]: This is how you can include headers in tests that are not in the public API. CMake lesson learned.

File: tests/SensorMathTesting.cpp [hunk 6]
```diff
@@ -7,50 +7,6 @@
...
-
-
-TEST(vecToCartesian, vec) {
-  arma::vec armaPoint{1.0, 2.0, 3.0}; 
-  CartesianVector cartesianVec = vecToCartesian(armaPoint);
-  EXPECT_DOUBLE_EQ(1.0, cartesianVec.x);
-  EXPECT_DOUBLE_EQ(2.0, cartesianVec.y);
-  EXPECT_DOUBLE_EQ(3.0, cartesianVec.z);
-}
-
```
Comment (jlaura) [inline tests/SensorMathTesting.cpp]: I would argue that these do not actually test anything that we should be testing. This is checking that the arma vec is being packed into a struct. This seems like something that has the potential to be more of a burden to maintain tests. What happens if we drop armadillo? Open to the other side though that we should be testing these.

File: api/SensorModel.h (hunk 2 above)
Comment (jessemapel) [reply api/SensorModel.h]: It should be included but it needs to be virtual. c++ gets real weird if your base class doesn't have a virtual destructor.

File: tests/SensorMathTesting.cpp (hunk 6 above)
Comment (jessemapel) [reply tests/SensorMathTesting.cpp]: So these are testing a super simple wrapper around the Armadillo constructors. I agree that these test will cause maintainability problems in the future. I don't like having functionality indirectly tested, but this functionality is so simple.

**Pull request merged**